src/backend/distributed/operations/worker_split_copy_udf.c

```diff
@@ -0,0 +1,261 @@
+/*-------------------------------------------------------------------------
+ *
+ * worker_split_copy_udf.c
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "pg_version_compat.h"
+#include "utils/lsyscache.h"
+#include "utils/array.h"
+#include "utils/builtins.h"
+#include "distributed/listutils.h"
+#include "distributed/multi_executor.h"
+#include "distributed/worker_shard_copy.h"
+#include "distributed/intermediate_results.h"
+#include "distributed/citus_ruleutils.h"
+
+PG_FUNCTION_INFO_V1(worker_split_copy);
+
+typedef struct SplitCopyInfo
+{
+	uint64 destinationShardId;              /* destination shard id */
+	Datum destinationShardMinHashValue;     /* min hash value of destination shard */
+	Datum destinationShardMaxHashValue;     /* max hash value of destination shard */
+	uint32_t destinationShardNodeId;        /* node where split child shard is to be placed */
+} SplitCopyInfo;
+
+static void ParseSplitCopyInfoDatum(Datum splitCopyInfoDatum,
+									SplitCopyInfo **splitCopyInfo);
+static DestReceiver ** CreateShardCopyDestReceivers(EState *estate,
+													ShardInterval *
+													shardIntervalToSplitCopy,
+													List *splitCopyInfoList);
+static DestReceiver *  CreatePartitionedSplitCopyDestReceiver(EState *executor,
+															  ShardInterval *
+															  shardIntervalToSplitCopy,
+															  List *splitCopyInfoList);
+static void BuildMinMaxRangeArrays(List *splitCopyInfoList, ArrayType **minValueArray,
+								   ArrayType **maxValueArray);
+
+/*
+ * worker_split_copy(source_shard_id bigint, splitCopyInfo citus.split_copy_info[])
+ * UDF to split copy shard to list of destination shards.
+ * 'source_shard_id' : Source ShardId to split copy.
+ * 'splitCopyInfos'   : Array of Split Copy Info (destination_shard's id, min/max ranges and node_id)
+ */
+Datum
+worker_split_copy(PG_FUNCTION_ARGS)
+{
+	uint64 shardIdToSplitCopy = DatumGetUInt64(PG_GETARG_DATUM(0));
+	ShardInterval *shardIntervalToSplitCopy = LoadShardInterval(shardIdToSplitCopy);
+
+	ArrayType *splitCopyInfoArrayObject = PG_GETARG_ARRAYTYPE_P(1);
+	if (array_contains_nulls(splitCopyInfoArrayObject))
+	{
+		ereport(ERROR,
```

Comment:
According to codecov this line is not covered. It should be easy to a test case that covers this afaict.